docs/sql-programming-guide.md

```diff
@@ -1073,6 +1073,16 @@ the following case-sensitive options:
   </tr>
 
   <tr>
+     <td><code>numPartitions</code></td>
+     <td>
+       The number of partitions that can be used, if set. It works by limiting both read and write
+       operations' parallelism. If the number of partitions to write exceeds this limit, the
+       operation will coalesce the data set to fewer partitions before writing. In other words,
```

Comment:
In the read path, we might not generate the exact number of partitions.

Found a bug in the read path.